charts/consul/templates/api-gateway-gatewayclassconfig.yaml

```diff
@@ -15,6 +15,9 @@ spec:
       {{- if .Values.global.acls.manageSystemACLs }}
       managed: true
       method: {{ template "consul.fullname" . }}-k8s-auth-method
+      {{- if .Values.global.enablePodSecurityPolicies }}
+      podSecurityPolicy: {{ template "consul.fullname" . }}-api-gateway
```

Comment:
This specifies the name of the `podSecurityPolicy` that should apply to any `Gateway` using this `GatewayClassConfig`. The API gateway controller will then create a `Role` and `RoleBinding` to bind this PSP to the `ServiceAccount` that it creates.

This is intentionally pretty flexible so that a user doesn't need a code release from us if they, for example, wanted to create their own `PodSecurityPolicy` and have the API gateway controller use that. By allowing this, we let users tailor to their needs without requiring a code release from us.